src/core/request_client.rs

```diff
@@ -29,6 +30,27 @@ pub struct RequestClient {
 }
 
 impl RequestClient {
+    /// Creates a new `RequestClient`, bootstrapping the socket
+    /// location.
+    pub fn new() -> Result<Self> {
+        if env::var("PARSEC_SERVICE_ENDPOINT").is_ok() {
+            Ok(RequestClient {
+                ipc_handler: ipc_handler::connector_from_url(url::Url::parse(
+                    &env::var("PARSEC_SERVICE_ENDPOINT").unwrap(),
+                )?)?,
+                max_body_size: DEFAULT_MAX_BODY_SIZE,
+            })
+        } else {
+            Ok(RequestClient {
+                max_body_size: DEFAULT_MAX_BODY_SIZE,
+                ipc_handler: Box::from(unix_socket::Handler::new(
+                    unix_socket::DEFAULT_SOCKET_PATH.into(),
+                    Some(ipc_handler::DEFAULT_TIMEOUT),
+                )?),
+            })
+        }
```

Comment:
instead of this `else` branch you could leverage the constructor from URL and do:
```rust
let url = env::var("PARSEC_SERVICE_ENDPOINT").or(DEFAULT_SERVICE_ENDPOINT);
```
and only keep the `Ok` branch.

and have `DEFAULT_SERVICE_ENDPOINT` as `unix:///run/parsec/parsec.sock`

Reply:
Done. I didn't create a `DEFAULT_SERVICE_ENDPOINT` mostly because it's not that straight forward to combine two `const &str` to obtain a new one, so I just stuck to formatting `DEFAULT_SOCKET_PATH` in